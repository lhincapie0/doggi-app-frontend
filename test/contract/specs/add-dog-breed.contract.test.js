import { Matchers } from "@pact-foundation/pact";
import { getProvider } from "./init-pact";
import dogsApi from "../../../src/client/DogsApi";
import { createDogBreedMock, dogBreedResponse } from "../helpers/mockData";

const provider = getProvider();

describe("Given a Dog breed service", () => {
  beforeAll(async () => {
    console.log("Setting up provider...");
    await provider.setup();
  });

  describe("When a request to create a new dog breed is made", () => {
    beforeAll(async () => {
      console.log("Setting up env for create dog breed test...");

      await provider.addInteraction({
        uponReceiving: "A request to create a new dog breed",
        state: "create new dog breed",
        withRequest: {
          method: "POST",
          path: "/api/v1.0/dogbreeds/",
          body: createDogBreedMock,
        },
        willRespondWith: {
          status: 201,
          body: Matchers.like(dogBreedResponse),
        },
      });
    });

    it("Then it should return the right data", async () => {
      const createDogBreedResponse = await dogsApi.postDogBreed(
        createDogBreedMock
      );
      expect(createDogBreedResponse.data).toMatchSnapshot();
      provider.verify();
    });
  });

  afterAll(async () => {
    await provider.finalize();
  });
});
