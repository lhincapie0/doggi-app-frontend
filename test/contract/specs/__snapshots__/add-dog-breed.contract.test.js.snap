// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Dog breed service When a request to create a new dog breed is made Then it should return the right data 1`] = `
Object {
  "country": "Reino Unido",
  "dogBreedColors": Array [
    Object {
      "id": 1,
      "name": "Red",
    },
    Object {
      "id": 2,
      "name": "Black",
    },
    Object {
      "id": 3,
      "name": "Brindle",
    },
  ],
  "dogBreedNatures": Array [
    Object {
      "id": 1,
      "name": "sweet-tempered",
    },
    Object {
      "id": 2,
      "name": "Trainable",
    },
    Object {
      "id": 3,
      "name": "Stubborn",
    },
  ],
  "height": 31,
  "id": 1,
  "lifeExpectancy": 9,
  "name": "Beagle",
  "weight": 35,
}
`;
